src/DocLinkChecker/Program.cs

```diff
@@ -455,16 +455,36 @@ private static void ProcessFile(DirectoryInfo folder, string filepath)
                                                 // Get rid of the title mark
                                                 var lineTitleLink = lineTitle.Replace("#", string.Empty);
 
+                                                // check if there is an HTML link ID like this: ## <a id=\"i_am_an_id\">Title for ID</a>
+                                                Regex idLinkPattern = new Regex(@"(?><a id=\"")(?<id_label>[a-zA-Z -_]+)(?>\"">)");
+                                                if (idLinkPattern.IsMatch(lineTitleLink))
+                                                {
+                                                    // we have a match for this pattern!
+
+                                                    // check if the label id matches
+                                                    if (idLinkPattern.Match(lineTitleLink).Groups["id_label"].Value == afterSharp)
+                                                    {
+                                                        found = true;
+                                                        break;
+                                                    }
+
+                                                    // proceed with replacing the HTML link ID
+                                                    lineTitleLink = idLinkPattern.Replace(lineTitleLink, string.Empty);
+
+                                                    // and the closing link tag too
+                                                    lineTitleLink = lineTitleLink.Replace("</a>", string.Empty);
+                                                }
+
                                                 // Remove the space
                                                 lineTitleLink = lineTitleLink.TrimStart();
 
                                                 // To lower
                                                 lineTitleLink = lineTitleLink.ToLower();
 
-                                                // Remove bold and italic as well as the . and few others
+                                                // Remove bold and italic as well as the . ? and few others
                                                 lineTitleLink = lineTitleLink.Replace("*", string.Empty).Replace(".", string.Empty).Replace("'", string.Empty)
-                                                    .Replace("\"", string.Empty).Replace("_", string.Empty).Replace("/", string.Empty).Replace("&", string.Empty)
-                                                    .Replace("(", string.Empty).Replace(")", string.Empty).Replace("`", string.Empty);
+                                                    .Replace("\"", string.Empty).Replace("/", string.Empty).Replace("&", string.Empty)
+                                                    .Replace("(", string.Empty).Replace(")", string.Empty).Replace("`", string.Empty).Replace("?", string.Empty);
```

Comment:
This is a valid one and actually a bug! Do you we are missing the question mark here! If you mind reopening that one, it will fix one of the bug!

Reply:
![image](https://user-images.githubusercontent.com/8145578/207369366-0ba8b024-2015-4026-8d09-cd7229ea1082.png)

and this was not properly catch:
![image](https://user-images.githubusercontent.com/8145578/207369500-95068ec5-28f1-4326-bfff-0ecfeef4951a.png)

Reply:
Yes, it seems that no punctuation marks are supposed to be used in titles. So we're good.

Reply:
Yes, it seems that ? it's allowed in titles. So that's a bug.

https://github.com/DavidAnson/markdownlint/blob/v0.26.2/doc/Rules.md#md026---trailing-punctuation-in-heading

Reply:
Yes, that's correct, it's allowed! The bug is not the title but the tool :-D This is missing:
![image](https://user-images.githubusercontent.com/8145578/207388784-0fb4abd8-3388-443b-82e4-66e885871af7.png)